### PR TITLE
TileDB VFS api basic support

### PR DESCRIPTION
**Pull request by jakebolewski**
Loosely follows java.nio.Files naming scheme

### CODE REVIEW

**Comment by jakebolewski** [comment]
Have WIP Input/Output stream subclasses but punted for now as they needed more testing.

File: src/test/java/io/tiledb/java/api/VFSTest.java [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+  public void testVFSCreateDirectory() throws Exception {
+    try (Context ctx = new Context();
+        VFS vfs = new VFS(ctx)) {
+      String dirPath = Paths.get(tmp.getRoot().toString(), "foo").toString();
+      String newDir = vfs.createDirectory(dirPath);
+      Assert.assertTrue(vfs.isDirectory(newDir));
+    }
+  }
+
+  /* @Test(expected = TileDBError.class)
```
Comment (Shelnutt2) [inline src/test/java/io/tiledb/java/api/VFSTest.java]: Is this test suppose to be commented out?

File: src/test/java/io/tiledb/java/api/VFSTest.java [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+        VFS vfs = new VFS(ctx)) {
+      String filePath = Paths.get(tmp.getRoot().toString(), "bar").toString();
+      URI fileURI = URI.create("file://".concat(filePath));
+      vfs.createFile(fileURI);
+      Assert.assertTrue(vfs.isFile(fileURI));
+    }
+  }
+
+  /*  @Test(expected = TileDBError.class)
+      public void testCreateFileExistsErrors() throws Exception {
```
Comment (Shelnutt2) [inline src/test/java/io/tiledb/java/api/VFSTest.java]: Is this test suppose to be commented out?

File: src/test/java/io/tiledb/java/api/VFSTest.java [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+  }
+
+  @Test
+  public void testVFSFileSize() throws Exception {
+    try (Context ctx = new Context();
+        VFS vfs = new VFS(ctx)) {
+      String filePath = Paths.get(tmp.getRoot().toString(), "bar").toString();
+      vfs.createFile(filePath);
+      Assert.assertEquals(vfs.fileSize(filePath), 0L);
+      // TODO:
```
Comment (Shelnutt2) [inline src/test/java/io/tiledb/java/api/VFSTest.java]: Does the temp file "bar" need to be cleaned up here? What is the TODO?

File: src/test/java/io/tiledb/java/api/VFSTest.java (hunk 1 above)
Comment (jakebolewski) [reply src/test/java/io/tiledb/java/api/VFSTest.java]: Yeah we don't check on creation if the resource already exists, I thought we did. Removed

**Pull request merged**